http/get_simple/csharp/client/ArrowHttpClient.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Apache.Arrow;
+using Apache.Arrow.Ipc;
+
+namespace ArrowHttpClient
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            string serverUri = "http://127.0.0.1:8008/";
+
+            DateTime startTime = DateTime.UtcNow;
+
+            HttpClient httpClient = new HttpClient
+            {
+                BaseAddress = new Uri(serverUri),
+            };
+
+            using (var stream = await httpClient.GetStreamAsync(serverUri))
+            using (var reader = new ArrowStreamReader(stream))
+            {
+                Console.WriteLine("Connected");
+
+                Schema schema = reader.Schema;
```

Comment:
This line doesn't seem to be setting the schema. If I try to access it after, I get a `NullReferenceException`.

Reply:
This looks like a bug in the C# IPC reader where reader.Schema isn't set until after the first batch is read. I filed #40634 to track this.

Reply:
Thanks. In the meantime: I moved this line lower in #26.